src/mscorlib/src/System/Globalization/CompareInfo.Unix.cs

```diff
@@ -253,36 +253,44 @@ private unsafe int LastIndexOfCore(string source, string target, int startIndex,
             }
         }
 
-        private bool StartsWith(string source, string prefix, CompareOptions options)
+        private bool StartsWith(ReadOnlySpan<char> source, ReadOnlySpan<char> prefix, CompareOptions options)
         {
             Debug.Assert(!_invariantMode);
 
-            Debug.Assert(!string.IsNullOrEmpty(source));
-            Debug.Assert(!string.IsNullOrEmpty(prefix));
+            Debug.Assert(!source.IsEmpty);
+            Debug.Assert(!prefix.IsEmpty);
             Debug.Assert((options & (CompareOptions.Ordinal | CompareOptions.OrdinalIgnoreCase)) == 0);
 
             if (_isAsciiEqualityOrdinal && CanUseAsciiOrdinalForOptions(options) && source.IsFastSort() && prefix.IsFastSort())
```

Comment:
How should I deal with the fact that this method uses string.IsFastSort() which doesn't exist on Span?
https://github.com/dotnet/coreclr/blob/238907f69662dda53dee0afd9848756a5bb790e9/src/classlibnative/bcltype/stringnative.cpp#L173

Reply:
Refactor the implementation so that it is not needed. It will be likely faster without it, even for the string case: https://github.com/dotnet/coreclr/issues/14320